lib/restclient/request.rb

```diff
@@ -270,24 +278,25 @@ def make_headers user_headers
       headers
     end
 
-    # Do some sanity checks on cookie keys.
-    #
-    # Properly it should be a valid TOKEN per RFC 2616, but lots of servers are
-    # more liberal.
+    # Validate whether the cookie name contains invalid characters.
     #
     # Disallow the empty string as well as keys containing control characters,
     # equals sign, semicolon, comma, or space.
-    #
-    def valid_cookie_key?(string)
-      return false if string.empty?
+    # Properly it should be a valid TOKEN per RFC 2616, but lots of servers are
+    # more liberal.
+    # 
+    # @param cookie_name [String]
+    # @returns [Boolean]
+    def valid_cookie_key?(cookie_name)
+      return false if "#{cookie_name}".empty?
 
-      ! Regexp.new('[\x0-\x1f\x7f=;, ]').match(string)
+      ! Regexp.new('[\x0-\x1f\x7f=;, ]').match("#{cookie_name}")
     end
 
     # Validate cookie values. Rather than following RFC 6265, allow anything
     # but control characters, comma, and semicolon.
     def valid_cookie_value?(value)
-      ! Regexp.new('[\x0-\x1f\x7f,;]').match(value)
+      ! Regexp.new('[\x0-\x1f\x7f,;]').match("#{value}")
```

Comment:
I got an error when I tried to set a cookie like: `{my_cooke: 1234}`.  It was somewhat generic, like "TypeError: No implicit conversion of Fixnum into String".

This change brings cookie validation in line with the actual usage of that value above.  When the header is created interpolation is used:

```ruby
user_headers[:cookie] = @cookies.map { |key, val| "#{key}=#{val}" }.sort.join('; ')
```

Why not do that here as well?